src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -258,15 +258,6 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             // If we don't have a best completion item yet, then pick the first item from the list.
             var bestOrFirstCompletionItem = bestItem ?? itemsInList.First().FilterResult.CompletionItem;
 
-            // Check that it is a filter symbol. We can be called for a non-filter symbol.
-            if (filterReason == CompletionFilterReason.Insertion &&
-                !IsPotentialFilterCharacter(typeChar) &&
-                !string.IsNullOrEmpty(filterText) &&
-                !Helpers.IsFilterCharacter(bestOrFirstCompletionItem, typeChar, filterText))
-            {
-                return null;
-            }
```

Comment:
any concern about anything breaking if you do this?

Reply:
Thank you, @CyrusNajmabadi ! As I remember, this conditions were aggregated from various conditions in the old completion to support some corner cases not properly supported by Editor. This was caused by changing the order of Filter and Commit attempts in the new completion. 

Now it seems that those scenarios are properly covered on the Editor side and we do not need this exception anymore. Moreover, Editor asked us not to return null in the Update method.